internal/extract.go

```diff
@@ -202,6 +216,23 @@ func ExtractAllWithSleeper(tarInterpreter TarInterpreter, files []ReaderMaker, s
 	return nil
 }
 
+func extractFile(tarInterpreter TarInterpreter, extractingReader io.Reader, fileClosure ReaderMaker) error {
+	switch fileClosure.FileType() {
```

Comment:
I think this deserves a couple of comments, also I would rename `extractOne` to `extractOneTar` to avoid confusion.

Reply:
Rename and add comment